src/k-provider/ovp/provider.js

```diff
@@ -25,12 +26,25 @@ export default class OVPProvider extends BaseProvider<OVPProviderMediaInfoObject
     OVPConfiguration.set(options.env);
     this._setFilterOptionsConfig(options.filterOptions);
     this._networkRetryConfig = Object.assign(this._networkRetryConfig, options.networkRetryParameters);
+    this._anonymousKs = '';
   }
 
   get env() {
     return OVPConfiguration.get();
   }
 
+  get anonymousKs(): string {
+    return this._anonymousKs;
+  }
+
+  set anonymousKs(value: string): void {
+    this._anonymousKs = value;
```

Comment:
Why do you need setter? Are we planning to allow setting ks like this for anonymous? Or even user KS? I think we should only allow passing it via call and not separately like this.

Reply:
@OrenMe we need to save anonymous ks, this was the root cause for the issue. up until now, we only saved non-anonymous KS, and when api calls were then executed, ks was empty and calls failed. we created a different property to still distinguish between anonymous and non-anonymous ks. the userKS only has getter because in multiple places we check for this.ks || this.anonymousKS, so we just united them.

Reply:
no argue about having `_anonymousKs` but why do we need a public setter? are you planning to update it from outside the provider?